### ISSUE TITLE
[BUG(?)] Walkie-talkie filter is heard while close to the speaker & Turning off the walkie-talkie while someone is speaking does not interupt the audio

### ISSUE THREAD

**Issue by Fooxynn** [n33kos/AlwaysHearActiveWalkie#6]
So yeah i am having this mod with my friend as i thought it's a mandatory feature to be able to hear when close to any enabled walkie-talkie (WT). Tho we encountered 2 tiny issue that breack that perfect sensationof immersion a bit.

The first one is that when someone is speaking in a WT, if someone is close to him he will hear him like if he was receiving the transmission via radio (like with the filter effect of the WT). I am pretty sure that is unwanted.

The second one happen when you turn off the WT while someone is transmiting via it. You still can hear him as long as he don't stop speaking (like as long as audio is detected from his microphone). May be more or less accurate but at least for sure it don't deactivate the audio immediatly as it should.

**Comment by etrant** [comment]
This may be tied to how the mod handles holding WTs applying the right side filter and 2d spatial blend in the StartOfRound.UpdatePlayerVoiceEffects method. Working of fixing it for my group, should be able to PR if no one else gets it working first.


```
      // PlayerControllerBPatch
      // If the player is going in or out of range of an active walkie
      if (isAnyWalkieInRange != __instance.holdingWalkieTalkie) {
        // Set the holdingWalkieTalkie bool to true if the player is within range of an active walkie
        ___holdingWalkieTalkie = isAnyWalkieInRange;
        
        // ... Rest of the code
        
        }
        
        // UpdatePlayerVoiceEffects() snippet
        bool flag = playerControllerB2.speakingToWalkieTalkie && playerControllerB.holdingWalkieTalkie && playerControllerB2 != playerControllerB;
        // ...
        if (!flag)
		{
			currentVoiceChatAudioSource.spatialBlend = 1f;
			playerControllerB2.currentVoiceChatIngameSettings.set2D = false;
			currentVoiceChatAudioSource.bypassListenerEffects = false;
			currentVoiceChatAudioSource.bypassEffects = false;
			currentVoiceChatAudioSource.outputAudioMixerGroup = SoundManager.Instance.playerVoiceMixers[playerControllerB2.playerClientId];
			component.lowpassResonanceQ = 1f;
		}
		else
		{
			currentVoiceChatAudioSource.spatialBlend = 0f;
			playerControllerB2.currentVoiceChatIngameSettings.set2D = true;
			if (GameNetworkManager.Instance.localPlayerController.isPlayerDead)
			{
				currentVoiceChatAudioSource.panStereo = 0f;
				currentVoiceChatAudioSource.outputAudioMixerGroup = SoundManager.Instance.playerVoiceMixers[playerControllerB2.playerClientId];
				currentVoiceChatAudioSource.bypassListenerEffects = false;
				currentVoiceChatAudioSource.bypassEffects = false;
			}
			else
			{
			// Problematic code
				currentVoiceChatAudioSource.panStereo = 0.4f;
				currentVoiceChatAudioSource.bypassListenerEffects = false;
				currentVoiceChatAudioSource.bypassEffects = false;
				currentVoiceChatAudioSource.outputAudioMixerGroup = SoundManager.Instance.playerVoiceMixers[playerControllerB2.playerClientId];
			}
			component2.lowPassOverride = 4000f;
			component.lowpassResonanceQ = 3f;
		}
        
```

**Comment by n33kos** [comment]
@Fooxynn @wheatwarrior

1. I'm pretty sure this is resolved in version `1.4.3` I now have it trying to figure out which audio effect "source" would be louder and change between the two accordingly. 
2. I have been able to reproduce this issue but have been unable to solve it so far. I'll open up a new issue for that one so I can close this one out as resolved by 1.4.3
